src/main/java/org/spongepowered/common/event/filter/FilterGenerator.java

```diff
@@ -168,52 +175,58 @@ public static FilterGenerator getInstance() {
             if (sfilter != null) {
                 local = sfilter.write(name, cw, mv, method, local);
             }
-            for (FilterDelegate eventFilter : additional) {
+            for (final FilterDelegate eventFilter : additional) {
                 local = eventFilter.write(name, cw, mv, method, local);
             }
 
             // local var indices of the parameters values
-            int[] plocals = new int[params.length - 1];
-            for (int i = 1; i < params.length; i++) {
-                Parameter param = params[i];
+            final int[] plocals = new int[parameters.length - 1];
+            for (int i = 1; i < parameters.length; i++) {
+                final ListenerClassVisitor.ListenerParameter param = parameters[i];
                 ParameterFilterSourceDelegate source = null;
-                List<ParameterFilterDelegate> paramFilters = new ArrayList<>();
-                for (Annotation anno : param.getAnnotations()) {
-                    Object obj = FilterGenerator.filterFromAnnotation(anno.annotationType());
+                final List<ParameterFilterDelegate> paramFilters = new ArrayList<>();
+                for (final ListenerClassVisitor.ListenerAnnotation anno : param.annotations()) {
+                    final Object obj = FilterGenerator.filterFromAnnotation(Class.forName(anno.type().getClassName()));
```

Comment:
This should probably use `method.declaringClass().getClassLoader()` to look up the annotation class name